exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +74,2808 @@ func TestStart(t *testing.T) {
+		t.Errorf("%s GetSubAccounts() error %v", g.Name, err)
+	}
+}
+
+func TestGetSingleSubAccount(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetSingleSubAccount(context.Background(), "Sub_Acconunt_for_testing"); err != nil && !strings.Contains(err.Error(), "FORBIDDEN") {
+		t.Errorf("%s GetSingleSubAccount() error %v", g.Name, err)
+	}
+}
+
+// Wrapper test functions
+
+func TestFetchTradablePairs(t *testing.T) {
+	t.Parallel()
+	_, err := g.FetchTradablePairs(context.Background(), asset.DeliveryFutures)
+	if err != nil {
+		t.Errorf("%s FetchTradablePairs() error %v", g.Name, err)
+	}
+	if _, err = g.FetchTradablePairs(context.Background(), asset.Options); err != nil {
+		t.Errorf("%s FetchTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTradablePairs(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTradablePairs(context.Background(), true); err != nil {
+		t.Errorf("%s UpdateTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTickers(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTickers(context.Background(), asset.DeliveryFutures); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp, err := getFirstTradablePair(t, asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := g.UpdateOrderbook(context.Background(), cp, asset.Futures); err != nil {
+		t.Errorf("%s UpdateOrderbook() error %v", g.Name, err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", g.Name, err)
+	}
+}
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	currencyPair := currency.NewPair(currency.BTC, currency.USDT)
+	_, err := g.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	cp, err := getFirstTradablePair(t, asset.DeliveryFutures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), cp, asset.DeliveryFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		ClientID:  "meowOrder",
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
```

Comment:
Text field is not required and should be generated before Submit call. Can remove that. Also it doesn't trasnlate back to `clientOrderID`

Raw response: 

```json
{"id":"246879011262","text":"t-7847449558","create_time":"1671412003","update_time":"1671412003","create_time_ms":1671412003210,"update_time_ms":1671412003210,"status":"open","currency_pair":"MNW_USDT","type":"limit","account":"spot","side":"sell","amount":"1","price":"150","time_in_force":"gtc","iceberg":"0","left":"1","fill_price":"0","filled_total":"0","fee":"0","fee_currency":"USDT","point_fee":"0","gt_fee":"0","gt_maker_fee":"0","gt_taker_fee":"0","gt_discount":false,"rebated_fee":"0","rebated_fee_currency":"MNW"}
```

Returned type with missing fields (please check)
```
{Exchange:GateIO Type:LIMIT Side:SELL Pair:mnw_usdt AssetType:spot ImmediateOrCancel:false FillOrKill:false PostOnly:false ReduceOnly:false Leverage:0 Price:150 Amount:1 QuoteAmount:0 TriggerPrice:0 ClientID:meowOrder ClientOrderID: LastUpdated:2022-12-19 12:06:43.2892507 +1100 AEDT m=+1.726163901 Date:2022-12-19 12:06:43 +1100 AEDT Status:OPEN OrderID:246879011262 Trades:[] Fee:0 Cost:0}
```